database/database_test.go

```diff
@@ -0,0 +1,645 @@
+// Package database_test contains white box test cases for database session
+// interface.
+package database_test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/deckarep/golang-set"
+	"github.com/pborman/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/coreos/clair/database"
+	"github.com/coreos/clair/pkg/pagination"
+
+	// register database implementations
+	_ "github.com/coreos/clair/database/pgsql"
+)
+
+// testSession is an interface only for testing purpose in order to look into
+// the actual database.
+type testSession interface {
+	TGetAllDetectors() ([]database.Detector, error)
+	TGetAllNotifications() ([]database.NotificationHook, error)
+	TGetAllNamespaces() ([]database.Namespace, error)
+	TGetAllFeatures() ([]database.Feature, error)
+	TGetAllNamespacedFeatures() ([]database.NamespacedFeature, error)
+	TGetAllVulnerabilitiesWithAffected() ([]database.VulnerabilityWithAffected, []bool, error)
+	TGetAllLock() ([]database.Lock, error)
+}
+
+type mockPage struct {
```

Comment:
This is the implementation I'm dubious about. 
https://github.com/coreos/clair/blob/master/database/pgsql/notification.go#L214 
Here notification.go expects the Page struct to be exactly this, however, this is an implicit requirement.
If any implementation uses any other Page struct implementation, we'll have trouble.